aspnetcore/blazor/security/blazor-web-app-with-entra.md

```diff
@@ -0,0 +1,172 @@
+* `{TENANT ID}`: The directory (tenant) ID.
+
+Example:
+
+```json
+"AzureAd": {
+  "CallbackPath": "/signin-oidc",
+  "ClientId": "41451fa7-82d9-4673-8fa5-69eff5a761fd",
+  "Domain": "contoso.onmicrosoft.com",
+  "Instance": "https://login.microsoftonline.com/",
+  "ResponseType": "code",
+  "TenantId": "e86c78e2-8bb4-4c41-aefd-918e0565a45e"
+},
+```
+
+The callback path (`CallbackPath`) must match the redirect URI (login callback path) configured when registering the application in the Entra or Azure portal. Paths are configured in the **Authentication** blade of the app's registration. The default value of `CallbackPath` is `/signin-oidc` for a registered redirect URI of `https://localhost/signin-oidc` (a port isn't required).
+
+<!-- UPDATE 9.0 Replace the following with the INCLUDE link after 
+                https://github.com/dotnet/AspNetCore.Docs/pull/33749
+                merges. 
+                
+                [!INCLUDE[](~/blazor/security/includes/secure-authentication-flows.md)]
+                -->
+
+> [!WARNING]
+> Don't store app secrets, connection strings, credentials, passwords, personal identification numbers (PINs), private C#/.NET code, or private keys/tokens in client-side code, which is ***always insecure***. In test/staging and production environments, server-side Blazor code and web APIs should use secure authentication flows that avoid maintaining credentials within project code, configuration files, and environment variables.
+
+## Redirect to the home page on sign out
+
+When a user navigates around the app, the `LogInOrOut` component (`Layout/LogInOrOut.razor`) sets a hidden field for the return URL (`ReturnUrl`) to the value of the current URL (`currentURL`). When the user signs out of the app, the identity provider returns them to the page from which they signed out.
+
+If the user signs out from a secure page, they're returned back to the same secure page after signing out only to be sent back through the authentication process. This behavior is fine when users need to switch accounts frequently. However, a alternative app specification may call for the user to be returned to the app's home page or some other page after signing out. The following example shows how to set the app's home page as the return URL for sign-out operations.
+
+The important changes to the `LogInOrOut` component are demonstrated in the following example. The `value` of the hidden field for the `ReturnUrl` is set to the home page at `/`. <xref:System.IDisposable> is no longer implemented. The <xref:Microsoft.AspNetCore.Components.NavigationManager> is no longer injected. The entire `@code` block is removed.
+
+`Layout/LogInOrOut.razor`:
+
+```razor
+@using Microsoft.AspNetCore.Authorization
+
+<div class="nav-item px-3">
+    <AuthorizeView>
+        <Authorized>
+            <form action="authentication/logout" method="post">
+                <AntiforgeryToken />
+                <input type="hidden" name="ReturnUrl" value="/" />
```

Comment:
We could even remove the above line, since [`/` is the default value](https://github.com/dotnet/blazor-samples/blob/7bbc65ae1f8d636fcf8e4934133b82c1b72fe10d/9.0/BlazorWebAppEntra/BlazorWebAppEntra/LoginLogoutEndpointRouteBuilderExtensions.cs#L33), but I understand leaving it for explanatory purposes.

Reply:
In that case, I'll remove it and mention it in the text.